### PR TITLE
UCS-120 : pip notifies RM instead of CH

### PR DESCRIPTION
**Pull request by hexec**

### CODE REVIEW

File: PIPReader/src/main/java/it/cnr/iit/ucs/pipreader/PIPReaderSubscriberTimer.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
     private Timer timer;
     PIPReader pip;
 
-    private static final long DEFAULT_RATE = 3L * 1000;
+    private static final long DEFAULT_RATE = 1L * 1000;
```
Comment (hexec) [inline PIPReader/src/main/java/it/cnr/iit/ucs/pipreader/PIPReaderSubscriberTimer.java]: faster! :)

File: UCSLibraries/src/main/java/it/cnr/iit/ucs/requestmanager/RequestManagerInterface.java [hunk 2]
```diff
@@ -25,17 +25,16 @@
...
     /**
-     * Sends a message from the external world to the context handler
+     * Receives a message from the external world
      *
      * @param message
      *          the message to be sent
      * @return the response of the ContextHandler
      */
-    public boolean sendMessageToCH( Message message );
+    public boolean sendMessage( Message message );
```
Comment (hexec) [inline UCSLibraries/src/main/java/it/cnr/iit/ucs/requestmanager/RequestManagerInterface.java]: when we use requestManager.sendMessage(message) we are not using the rm to send the message right away. We are actually receiving a message to be then stored or forwarded to the contextHandler.
Would it be more logical to call this function requestManager.putMessage or .receiveMessage ?

Comment (tezcan10) [reply UCSLibraries/src/main/java/it/cnr/iit/ucs/requestmanager/RequestManagerInterface.java]: +1
that would make the purpose clearer

File: PIPReader/src/main/java/it/cnr/iit/ucs/pipreader/PIPReader.java [hunk 3]
```diff
@@ -333,6 +333,7 @@ public void notifyContextHandler( Attribute attribute ) {
         AttributeChangeMessage attrChangeMessage = new AttributeChangeMessage( ENTITIES.PIP.toString(), ENTITIES.CH.toString() );
         ArrayList<Attribute> attrList = new ArrayList<>( Arrays.asList( attribute ) );
         attrChangeMessage.setAttributes( attrList );
-        getContextHandler().attributeChanged( attrChangeMessage );
+        log.info( "!!! pip->rm" );
```
Comment (antoniolm23) [inline PIPReader/src/main/java/it/cnr/iit/ucs/pipreader/PIPReader.java]: please remove additional log

File: UCSCore/src/main/java/it/cnr/iit/ucs/requestmanager/RequestManager.java [hunk 4]
```diff
@@ -115,7 +115,11 @@ public Message call() {
 
     private void handleMessage( Message message ) throws Exception {
         Message responseMessage = null;
-        if( message.getPurpose() == PURPOSE.TRY ) {
+        if( message instanceof AttributeChangeMessage ) {
+            log.info( "!!! attrchange" );
```
Comment (antoniolm23) [inline UCSCore/src/main/java/it/cnr/iit/ucs/requestmanager/RequestManager.java]: additional log also here

**Pull request merged**